### ISSUE TITLE
libGL error: MESA-LOADER: failed to open radeonsi with Proton 5.13-1

### ISSUE THREAD

**Issue by AwesamLinux** [ValveSoftware/steam-runtime#309]
## System Information
- GPU: RX 5700
- GPU Driver: Mesa 20.3.0-devel
- Kernel version: 5.8.16-050816-lowlatency
- Distro version: Mint 20 (64 bit) (Cinnamon)
- Proton version: 5.13-1

## Symptoms 

Most games I have tested are no longer launching with Proton 5.13-1, these are games that I tried and still works for me with Proton 5.0-9.

_I think I have the same/related issue as ValveSoftware/Proton#4278 and ValveSoftware/Proton#4269, however those report says **can't run anything** but that is not the case for me, therefore I'm posting this a separate issue just in case._

I included a bunch of logs trying to launch games with Proton 5.13-1. All these games do still work for me with Proton 5.0-9 but don't launch with 5.13-1 (nothing happens). I also included some logs of games that did launch for me with 5.13-1 for comparison.

**FAILED TO LAUNCH:**
steam-1349960   The Ghost Train - FAIL
steam-501300    What Remains of Edith Finch - FAIL
steam-1057750   The Suicide of Rachel Foster - FAIL
steam-343710    Kholat - FAIL
steam-1122720   Sayonara Wild Hearts - FAIL
steam-752590    A Plague Tale: Innocence - FAIL
steam-826940    Maid of Sker - FAIL

**SUCCESSFULLY LAUNCHED:**
steam-368430    Through the Woods - SUCCESS
steam-217920    Alien Rage - SUCCESS
steam-321270    UNLOVED - SUCCESS

**What I have tried:**
- Verifying Proton files through steam
- Deleting the Proton folders manually, and verifying/letting Steam downloading them again.

**Logs:** 
Logs of launching all these games with Proton 5.13-1

[logs.zip](https://github.com/ValveSoftware/Proton/files/5396175/logs.zip)
[system_info.txt](https://github.com/ValveSoftware/Proton/files/5396177/system_info.txt)

Edit: System information after waiting for runtime info to be filled in:
[system_info.txt](https://github.com/ValveSoftware/Proton/files/5402268/system_info.txt)

**Update:**
I tried now also launching a few games with the  Proton 5.13 debug branch:
[logs_debug.zip](https://github.com/ValveSoftware/Proton/files/5396886/logs_debug.zip)

I have also tried verifying the "Steam Linux Runtime - Soldier" files, it is curiously for me listed as a game not a tool **(that does not seem right?)**

**Comment by TheAquabat** [comment]
I'm not sure if it is the same bug proton 5.13-2 doesn't work, but Proton 5.21 GE (which also uses pressure vessel does) I will investigate further and report back.

**Comment by smcv** [comment]
> I'm not sure if it is the same bug

If you're not sure, report it separately. Having the full information is useful, and we can close a separate report as a duplicate more easily than we can disentangle two bugs being mixed up in the same issue number.

**Comment by TheAquabat** [comment]
yes can confirm that removing amdgpu pro libdrm libs fixes the issue but only for Proton 5.21 GE and not for standard proton 5.13-2
When I mean amdgpu pro libdrm libs I mean the driver that you can download from amd.com I use it in headless mode installing the driver like this
sudo ./amdgpu-pro-install --opencl=legacy,pal --headless

to use only the OpenCL driver


here's a container https://gist.github.com/TheAquabat/68ad6d803d39170afdc18bb2e75b3f4f

here's steam system info https://gist.github.com/TheAquabat/9ab48978c5666dd5e17688efcc5f78c6

this is my system 

NAME="KDE neon"
VERSION="5.20"
ID=ubuntu
ID_LIKE="ubuntu debian"
PRETTY_NAME="KDE neon User Edition 5.20"
VARIANT="User Edition"
VERSION_ID="20.04"
HOME_URL="https://neon.kde.org/"
SUPPORT_URL="https://neon.kde.org/"
BUG_REPORT_URL="https://bugs.kde.org/"
LOGO=start-here-kde-neon
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal

**Comment by TheAquabat** [comment]
and I also have the amdvlk deb package installed https://github.com/GPUOpen-Drivers/AMDVLK

**Comment by TheAquabat** [comment]
I think that the problem might be the combination of both amdvlk and the amdgpu pro headless libs... not sure if it is the same bug. Should I open a new bug report?

**Comment by smcv** [comment]
> here's a container [log] https://gist.github.com/TheAquabat/68ad6d803d39170afdc18bb2e75b3f4f

Thanks. This looks better than we had before the last attempt to fix this: we're picking up at least some of the necessary libraries from `/opt`, which we weren't before.

> i386-linux-gnu-check-vulkan: symbol lookup error: /overrides/lib/i386-linux-gnu/vulkan/5/libvulkan_radeon.so: undefined symbol: amdgpu_cs_syncobj_transfer

That symbol is meant to be in `libdrm.so.2` or maybe `libdrm_amdgpu.so.1`, so this looks like we're ending up with a wrong version of libdrm. Apparently the version we're loading now is from `/opt/amdgpu-pro/lib/x86_64-linux-gnu`.

> /overrides/lib/i386-linux-gnu/dri/radeonsi_dri.so: undefined symbol: amdgpu_cs_query_reset_state2

That looks similar.

We also see a similar story for x86_64.

**Comment by smcv** [comment]
OK, this is weird. In https://gist.github.com/TheAquabat/9ab48978c5666dd5e17688efcc5f78c6 - both when we run directly on the host and when we run in the soldier container - we pick up the complete graphics stack from `/usr`, which seems to be the good one with newer versions of libraries. In particular, we get both `libdrm_amdgpu.so.1` and `libOpenCL.so.1` from `/usr`.

But then when you launch from outside Steam, we're somehow deciding that `libdrm_amdgpu.so.1` and `libOpenCL.so.1` from `/opt/amdgpu-pro` are the ones we should use: this means the library search path is being searched in a different order?

Did you get the container log and the system info at the same time; or did you do one, then add or remove workarounds, then do the other?

> When I mean amdgpu pro libdrm libs I mean the driver that you can download from amd.com

Please be much more specific than this: I need to see version numbers and URLs if I'm going to have any chance of understanding what is happening here.

**Comment by smcv** [comment]
Looking at, for example, https://www.amd.com/en/support/graphics/amd-radeon-5700-series/amd-radeon-rx-5700-series/amd-radeon-rx-5700 -> Ubuntu x86 64-Bit -> https://drivers.amd.com/drivers/linux/amdgpu-pro-20.45-1164792-ubuntu-20.04.tar.xz as a random example of something that someone might download:

This is looking a lot like a bug in the amdgpu-pro driver. The driver package claims to have been prepared specifically for Ubuntu 20.04, but it contains `libdrm_amdgpu.so.1` version 2.4.100, which is older than the `libdrm_amdgpu.so.1` version 2.4.102 in Ubuntu. Driver vendors can't just downgrade libraries and expect the resulting system to continue to work.

Until this is resolved, I would recommend getting AMD drivers from the OS vendor (in your case Ubuntu) instead of directly from AMD.

**Comment by smcv** [comment]
https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/-/merge_requests/192 might help this, depending what is going on on the affected systems.
